extensions/amp-analytics/0.1/transport.js

```diff
@@ -118,13 +152,215 @@ export class Transport {
+    assuredNonNullFrameData.newCreativeMessageQueue.enqueue(
+      this.id_, opt_extraData);
+    return assuredNonNullFrameData;
+  }
+
+  /**
+   * Called when a creative no longer needs its cross-domain iframe (for
+   * instance, because the creative has been removed from the DOM).
+   * Once all creatives using a frame are done with it, the frame can be
+   * destroyed.
+   * @param {!HTMLDocument} ampDoc The AMP document
+   * @param {!Object<string,string>} transportOptions The 'transport' portion
+   * of the amp-analytics config object
+   */
+  static doneUsingCrossDomainIframe(ampDoc, transportOptions) {
+    const frameUrl = transportOptions['iframe'];
+    const frameData = Transport.getFrameData_(frameUrl);
+    if (!frameData) {
+      // Didn't exist
+      return;
+    }
+    if (Transport.decrementIframeUsageCount_(
+      /** @type{!FrameData} */ (frameData)) > 0) {
+      // Some other instance is still using it
+      return;
+    }
+    ampDoc.body.removeChild(frameData.frame);
+    delete Transport.crossDomainIframes_[frameUrl];
+  }
+
+  /**
+   * Record that one more creative is using this cross-domain iframe
+   * @param {!FrameData} frameData  The cross-domain iframe and
+   * associated data
+   * @return {number}
+   * @private
+   */
+  static incrementIframeUsageCount_(frameData) {
+    return ++(frameData.usageCount);
+  }
+
+  /**
+   * Record that one fewer creative is using this cross-domain iframe
+   * @param {!FrameData} frameData  The cross-domain iframe and
+   * associated data
+   * @return {number}
+   * @private
+   */
+  static decrementIframeUsageCount_(frameData) {
+    return --(frameData.usageCount);
+  }
+
+  /**
+   * Returns whether a url of a cross-domain frame is already known
+   * @param {!string} frameUrl
+   * @return {!boolean}
+   * @private
+   */
+  static hasCrossDomainIframe_(frameUrl) {
+    return hasOwn(Transport.crossDomainIframes_, frameUrl);
+  }
+
+  /**
+   * Create a cross-domain iframe for third-party vendor anaytlics
+   * @param {!Window} win The window element
+   * @param {!string} frameUrl  The URL of the cross-domain iframe
+   * @return {!FrameData}
+   */
+  createCrossDomainIframe_(win, frameUrl) {
+    const sentinel = Transport.createUniqueId_();
+    const scriptSrc = getMode().localDev
+      ? '/dist.3p/current/ampanalytics-lib.js'
+      : `${urls.thirdParty}/$internalRuntimeVersion$/ampanalytics-v0.js`;
+    const frame = createElementWithAttributes(win.document, 'iframe', {
+      sandbox: 'allow-scripts',
+      name: JSON.stringify({
+        scriptSrc,
+        sentinel,
+      }),
+    });
+    const iframeMessagingClient = new IframeMessagingClient(window);
+    iframeMessagingClient.setSentinel(sentinel);
+    iframeMessagingClient.setHostWindow(
+      /** @type {!HTMLIFrameElement} */ (frame));
+    setStyles(frame,
+      {width: 0, height: 0, display: 'none',
+       position: 'absolute', top: 0, left: 0});
+    /** @const {FrameData} */
```

Comment:
I believe you want this to be:  const frameData = /** @type {!FrameData} */ ({....});